Use numba-cuda>=0.0.13

gmarkall opened the pull request:
## Description
Testing with https://github.com/NVIDIA/numba-cuda on CI.

I am not sure if edits in other repos are required (e.g. I used to have to change an "integration" repo) - any pointers appreciated!

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

msarahan commented:
This looks right from the recipe edit side. You should only have to manually edit dependencies.yaml, and all the other files should then be generated.

I don't understand the failing builds. I'm having a hard time finding the error that kills the build. Are you seeing this locally?

gmarkall commented:
Thanks @msarahan - unfortunately I edited the PR before I saw your comment - should I revert that last change? 

I haven't run this locally because I'm not really set up to build / test cudf at the moment (it's not what I usually work on), but will give it a try on Monday.

gmarkall commented:
Just noticed the problem, I introduced a space that shouldn't have been there.

msarahan commented on conda/recipes/cudf/meta.yaml:
```diff
@@ -82,7 +82,7 @@ requirements:
     - pandas >=2.0,<2.2.3dev0
     - cupy >=12.0.0
-    - numba >=0.57
+    - numba-cuda >= 0.0.13
```
extra space here is tripping it up, I think

```suggestion
    - numba-cuda >=0.0.13
```

gmarkall commented on dependencies.yaml:
```diff
@@ -565,7 +565,7 @@ dependencies:
         packages:
           - cachetools
-          - &numba numba>=0.57
+          - &numba_cuda numba-cuda>=0.0.13
```
IIUC this is assigning to `numba_cuda` - is that the best form for a name to avoid having a dash in a name, or for consistency with the package name should I change this back to:
```suggestion
          - &numba-cuda numba-cuda>=0.0.13
```

with corresponding change below?

msarahan replied:
I personally don't like using just plain package names as the anchor names, because it makes it harder to figure out  what to look/search for. I prefer things like `numba-cuda-dep`, so that the YAML anchor is distinct from the package name.

That might just be me, though, so I would advise you to follow your best judgement.

msarahan replied:
One other thing to keep in mind is that a YAML anchor can represent arbitrary chunks of YAML. It is not necessarily just one dependency. A naming formalism might help us with this ambiguity.

gmarkall replied:
I have no capacity for judgement here (I'm just cargo-cult changing things) so I'll go with your suggestion :-)

```suggestion
          - &numba-cuda-dep numba-cuda>=0.0.13
```

gmarkall commented on dependencies.yaml:
```diff
@@ -723,7 +723,7 @@ dependencies:
         packages:
           - dask-cuda==24.10.*,>=0.0.0a0
-          - *numba
+          - *numba_cuda
```
```suggestion
          - *numba-cuda-dep
```

gmarkall commented:
Thanks for looking at this @msarahan - I've just marked as "DO NOT MERGE" for now until I speak to Ben about this later in the Numba weekly sync - I want to first check whether there should be some discussion about the implications of this with other developers - will update here later.

gmarkall commented:
@msarahan Following the discussion in the cudf meeting last week the consensus seemed to be that this was OK to proceed with, so I've removed the "DO NOT MERGE" label, fixed up the conflicts, and I'm just waiting on CI.

vyasr commented:
@gmarkall what else do we want to see on this PR before we merge? If we're happy with it, I'd encourage moving forward ASAP so we have some decent runway to test it before the 24.10 release. CC @bdice @brandon-b-miller

gmarkall commented:
Thanks Vyas - I don't think there's anything else we want to see in it, I just wanted to have made sure to have discussed it with the cuDF team before merging.

vyasr commented:
Cool. Going to queue this up for merge then, the dep changes look fine.

vyasr commented:
/merge

vyasr commented:
Given the delay with the conda-forge recipe and there now being conflicts on this branch, let's push this upgrade to 24.12. That will give us more time to iron out any kinks if anything goes unexpectedly wrong with the switch (particularly if it happens downstream of cudf).

vyasr commented on dependencies.yaml:
```diff
@@ -720,7 +720,7 @@ dependencies:
           - matrix: {dependencies: "oldest"}
             packages:
-              - numba==0.57.*
+              - *numba-cuda-dep
```
Note that I changed this so that we now require numba-cuda even to test the oldest supported dependency. If at some point we choose to support a range of versions then we should remove the alias here and insert a real version.

vyasr commented:
/merge
